frontend/packages/ceph-storage-plugin/src/components/ocs-install/attached-devices/install-lso-sc.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import { match } from 'react-router';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch } from 'react-redux';
+import { ActionGroup, Button, Form } from '@patternfly/react-core';
+import {
+  NodeKind,
+  StorageClassResourceKind,
+  referenceForModel,
+  k8sCreate,
+  K8sResourceKind,
+} from '@console/internal/module/k8s';
+import {
+  withHandlePromise,
+  HandlePromiseProps,
+  history,
+  ButtonBar,
+} from '@console/internal/components/utils';
+import { setFlag } from '@console/internal/actions/features';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { getName } from '@console/shared';
+import { minSelectedNode, defaultRequestSize, NO_PROVISIONER } from '../../../constants';
+import { OCSServiceModel } from '../../../models';
+import AttachedDevicesNodeTable from './sc-node-list';
+import { PVsAvailableCapacity } from '../pvs-available-capacity';
+import { OCS_CONVERGED_FLAG, OCS_FLAG } from '../../../features';
+import { makeLabelNodesRequest } from '../create-form';
+import { LVSResource } from '../../../constants/resources';
+import { getOCSRequestData } from '../ocs-request-data';
+import {
+  OCSAlert,
+  SelectNodesSection,
+  StorageClassSection,
+} from '../../../utils/common-ocs-install-el';
+import { filterSCWithNoProv } from '../../../utils/install';
+import '../ocs-install.scss';
+import './attached-devices.scss';
+
+const makeOCSRequest = (
+  selectedData: NodeKind[],
+  storageClass: StorageClassResourceKind,
+): Promise<any> => {
+  const promises = makeLabelNodesRequest(selectedData);
+  const scName = getName(storageClass);
+  const ocsObj = getOCSRequestData(scName, defaultRequestSize.BAREMETAL, NO_PROVISIONER);
+
+  return Promise.all(promises).then(() => k8sCreate(OCSServiceModel, ocsObj));
+};
+
+export const CreateOCS = withHandlePromise<CreateOCSProps & HandlePromiseProps>((props) => {
+  const {
+    handlePromise,
+    errorMessage,
+    inProgress,
+    match: {
+      params: { appName, ns },
+    },
+  } = props;
+  const [filteredNodes, setFilteredNodes] = React.useState<string[]>([]);
+  const [storageClass, setStorageClass] = React.useState<StorageClassResourceKind>(null);
```

Comment:
Can you please explain what is reason for using `{}` and not `null`? I prefer initializing with null which is an empoty object reference

Reply:
The type `StorageClassResourceKind` is an object, therefore empty object makes sense here IMO

Reply:
null is an object too and null expresses a lack of identification, indicating that a variable points to no object which is what expected here.

Reply:
1. You need to have `storageClass?.{}`  like checks if you use `null`. 
2. The underlying component returns {} in case of no storage class. 
3. Initializing state with the matching type I would prefer.
Thats how I see :)

Reply:
As null is object too, so its of same type and here I want to initialize it with false value. Anyways its up to each preference and one should feel free to initialize what one thinks is best. `null` fits here so I will still prefer to initialize with null.

Reply:
ack